src/analogyFinder/src/models/Program.java

```diff
@@ -84,6 +84,18 @@ public Program reduce() {
 		if (codeBlocks == null) createCodeBlocks();
 		return codeBlocks;
 	}
+    
+    public List<Context> getReducedLocalContexts(List<Equivalence> equivalences) {
+        List<Context> reducedLocalContexts = new ArrayList<Context>();
+        for (int i = 0; i < tree.getPostorder().size() - 1; i++) {
+            Context context = tree.makeLocalContext(null, i);
+            for (Equivalence eq : equivalences) {
+                context.reduce(eq);
```

Comment:
This is no longer correct because reduce returns the reduced context rather than mutating the existing one.  We should probably name these various methods getReduced() rather than reduce() to emphasize this.